### ISSUE TITLE
Indent on last parenthesis even if it's closed

### ISSUE THREAD

**Issue by paradoxxxzero** [DSpeckhals/python-indent#15]
I get the following behavior:

``` python
 alpha = (
     beta(),#<newline>
          gamma
 )
```

which is obviously wrong since the last parenthesis is closed.

**Comment by DSpeckhals** [comment]
I see what you mean. It's trying to line up with the bracket on the first line. An example of this can be seen by modifying your example a bit:

``` python
 alpha = (
     epsilon(),#<newline>
          gamma
 )
```

To correct this, it needs pass this test in `python-indent.coffee`:

``` coffeescript
if lastOpenBracketLocations[0] < row - 1 and (not lastClosedRow.length or lastClosedRow[1] != row - 1)
        # The bracket was opened before the previous line,
        # and we did not just close a bracket,
        # we should use whatever indent we are given.
        # This will correctly handle continued hanging indents.
        return
```

Right now, it is failing that test because there _is_ a `lastClosedRow` _and_ `lastClosedRow[1]` is equal to the current `row`. I'm not exactly sure how to make this work yet (without parsing the entire file), but perhaps @kbrose can give some insight.

**Comment by kbrose** [comment]
Yes, the hanging indent parsing could use more work for sure. This is a good example, thanks for reporting @paradoxxxzero.

I'm thinking about two approaches:
1. Perhaps, in addition to storing the column where the bracket was opened in `openBracketStack`, a boolean indicating whether or not it is a hanging indent could be stored as well. This is not completely straight-forward, as whether or not a bracket is starting a hanging indent is not known until the rest of the line containing the bracket is parsed.
2. Alternatively, that block that @DSpeckhals could set the indentation to the level that was being used on the line containing the opening bracket corresponding to the last closed bracket. We would have to additionally make sure that the last closed bracket was _opened_ during the hanging indent, but that shouldn't be too hard...

I should be able to test out these ideas soon. Leaning towards (2) right now.

**Comment by DSpeckhals** [comment]
Fixed in PR #16.

**Comment by paradoxxxzero** [comment]
That's some nice reactivity! :+1:

**Comment by DSpeckhals** [comment]
The update has been published as version `0.4.1`.
